src/csm_testing/tests/sat_functional/firmware/test_firmware.py

```diff
@@ -0,0 +1,188 @@
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+import unittest
+import subprocess
+import shlex
+import re
+import os
+
+SAT_FIRMWARE_HEADER = ['xname', 'name', 'target_name', 'version']
+
+def get_xnames_from_firmware_output() -> list:
+    """Run 'sat firmware' and return a list of unique xnames from the output."""
+    command = "sat firmware"
+    proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+    output = proc.stdout.decode().strip()
+    lines = output.splitlines()
+    xnames = []
+    for line in lines[2:]:
+        fields = [col.strip() for col in line.split('|') if col.strip()]
+        if fields and fields[0].startswith('x') and fields[0] not in xnames:
+            xnames.append(fields[0])
+    return xnames
+
+class TestFirmware(unittest.TestCase):
+
+    def test_firmware_creates_and_deletes_snapshot(self):
+        """Test that running `sat firmware` creates a snapshot and outputs a report"""
+        command = "sat firmware"
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        output = proc.stdout.decode().strip()
+        info_msg = proc.stderr.decode().strip()
+
+        # Extract the snapshot name from the info message
+        pattern = (
+            r'INFO: Snapshot (\S+) created successfully\.\n'
+            r'INFO: Use `sat firmware --delete-snapshot` to delete the snapshot if it is no longer needed'
+        )
+        match = re.search(pattern, info_msg)
+        self.assertIsNotNone(match, "Snapshot creation message not found in info_msg.")
+        snapshot_name = match.group(1)
+
+        # Check that the output contains the expected table header
+        for col in SAT_FIRMWARE_HEADER:
+            self.assertIn(col, output)
+
+        # Now delete the snapshot to keep the system clean
+        delete_command = f"sat firmware --delete-snapshot {snapshot_name}"
+        proc = subprocess.run(shlex.split(delete_command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        delete_output = proc.stderr.decode().strip()
+        expected_msg = f"INFO: Snapshot {snapshot_name} deleted successfully."
+        self.assertIn(expected_msg, delete_output)
+
+    def test_firmware_describe_existing_snapshot(self):
+        """Test that `sat firmware --snapshots SNAPSHOT` describes an existing firmware snapshot."""
+        # Get the list of available snapshots
+        list_command = "sat firmware --snapshots"
+        proc = subprocess.run(shlex.split(list_command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        snapshots = [line.strip() for line in proc.stdout.decode().splitlines() if line.strip() and not line.startswith("#")]
+        self.assertGreater(len(snapshots), 0, "No firmware snapshots available for testing.")
```

Comment:
I would recommend trying to make the test methods more self-contained, so create a snapshot in this test (or in the `setUpClass` method, so it always exists for all test methods in this class and can be re-used). Then use that snapshot in this test method.